spanner/spanner_arrays/main.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Sample spanner_arrays is a basic program that queries Google's Cloud Spanner
+// and returns an array within the results.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"google.golang.org/api/iterator"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+// Country describes a country and the cities inside it
+type Country struct {
+	Name   string
+	Cities []spanner.NullString
+}
+
+func main() {
+	ctx := context.Background()
+
+	dsn := flag.String("database", "projects/your-project-id/instances/your-instance-id/databases/your-database-id", "Cloud Spanner database name")
+	flag.Parse()
+
+	// Connect to the Spanner Admin API
+	admin, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		log.Fatalf("failed to create database admin client: %v", err)
+	}
+	defer admin.Close()
+
+	err = createDatabase(ctx, admin, *dsn)
+	if err != nil {
+		log.Fatalf("failed to create database: %v", err)
+	}
+	defer removeDatabase(ctx, admin, *dsn)
+
+	// Connect to database
+	client, err := spanner.NewClient(ctx, *dsn)
+	if err != nil {
+		log.Fatalf("Failed to create client %v", err)
+	}
+	defer client.Close()
+
+	err = loadPresets(ctx, client)
+	if err != nil {
+		log.Fatalf("failed to load preset data: %v", err)
+	}
+
+	it := client.Single().Query(ctx, spanner.NewStatement(`
+		SELECT a.Name AS Name, ARRAY(
+			SELECT b.Name FROM Cities b WHERE a.CountryId = b.CountryId
+		) AS Cities FROM Countries a
+	`))
+	defer it.Stop()
+
+	for {
+		row, err := it.Next()
+		if err == iterator.Done {
+			return
+		}
+		if err != nil {
+			log.Fatalf("failed to read results: %v", err)
+		}
+
+		var country Country
+		err = row.ToStruct(&country)
```

Comment:
use compact error form:

`if err := row.ToStruct(&country); err != nil {`

Reply:
I rarely like to use the compact form, especially since it likes to cause vet-shadow errors, but I'm happy to accede to any norms in this case.